### ISSUE TITLE
Stuck and doing nothing for 40 minutes , ollama is not using Memory (RAM)

### ISSUE THREAD

**Issue by gokulcoder7** [TheBlewish/Automated-AI-Web-Researcher-Ollama#61]
I'm using Windows 11 OS
While app was running it used 70 to 99% of RAM 
But after 5 minutes , it stopped searching . 
ollama is not using Memory (RAM)

After 40 minutes still not searching

![IMG_20241128_104322_455](https://github.com/user-attachments/assets/ac5552f1-471b-43d6-9a6a-7ca6aa4edb9e)

**Comment by gokulcoder7** [comment]
Initializing system...
System initialized successfully.
📝 Enter your message (Press CTRL+Z to submit):
Discuss the necessity of digital service unit and channel service unit
Please start with '/' for search or '@' for research.
📝 Enter your message (Press CTRL+Z to submit):
/

Initiating web search...

Search attempt 1:
📝 Searching...
Original query:
Formulated query: Supreme Court rulings
Time range: y
⚙️ Scraping selected pages...
Successfully scraped: https://www.scotusblog.com/2024/06/announcement-of-opinions-for-friday-june-21/
Scraped content received for 1 URLs

Scraped Content:
URL: https://www.scotusblog.com/2024/06/announcement-of-opinions-for-friday-june-21/
Content: On Friday, June 21, we were live as the court released five opinions in cases from the current term. Click here for a list of FAQs about opinion announcements. Posted in Featured, Live Recommended Citation: SCOTUSblog , Announcement of opinions for Friday, June 21 (complete), SCOTUSblog (Jun. 21, 2024, 8:26 AM), https://www.scotusblog.com/2024/06/announcement-of-opinions-for-friday-june-21/...

🧠 Thinking...
Evaluation: The scraped content indicates that five opinions were released on Friday, June 21. However, it does not provide details about the cases themselves or their implications. For a comprehensive answer to the user's question, additional specifics are required regarding each case and its impact.
Decision: refine
Refining search...

Search attempt 2:
📝 Searching...
Original query:
Formulated query: Electric Vehicle Charging Station Costs
Time range: none

**Comment by leapingturtlefrog** [comment]
Given that you are essentially using 100% of your RAM, that may be the issue. Few questions:
- How much RAM do you have?
- Do you have a GPU? If so, what is it?
- Are you running the latest commit on the Windows branch [https://github.com/TheBlewish/Automated-AI-Web-Researcher-Ollama/tree/feature/windows-support](https://github.com/TheBlewish/Automated-AI-Web-Researcher-Ollama/tree/feature/windows-support)?

**Comment by gokulcoder7** [comment]
> Given that you are essentially using 100% of your RAM, that may be the issue. Few questions:
> 
> * How much RAM do you have?
> * Do you have a GPU? If so, what is it?
> * Are you running the latest commit on the Windows branch https://github.com/TheBlewish/Automated-AI-Web-Researcher-Ollama/tree/feature/windows-support?
16 GB RAM
Nvidia RTX 3060 GPU , BUT OLLAMA IS NOT USING GPU 

Yes I'm running latest commit

**Comment by gokulcoder7** [comment]
I pressed Q key many times to stop research but q is not displayed and research is still continuing .

**Comment by gokulcoder7** [comment]
@leapingturtlefrog please help

**Comment by nigelp** [comment]
Press q then CTRL-Z to activate the command.

**Comment by gokulcoder7** [comment]
> Press q then CTRL-Z to activate the command (CTRL-D on Linux)



I pressed q many times and it's not displayed

**Comment by gokulcoder7** [comment]
> @gokulcoder7 like nigelp said you have to submit the command with `ctrl+d` or 'ctrl+z', as seen here: ![image](https://private-user-images.githubusercontent.com/132923760/390889381-8f40c689-9c94-40d8-a45d-24720fd30072.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzI4NjU1MDEsIm5iZiI6MTczMjg2NTIwMSwicGF0aCI6Ii8xMzI5MjM3NjAvMzkwODg5MzgxLThmNDBjNjg5LTljOTQtNDBkOC1hNDVkLTI0NzIwZmQzMDA3Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyOVQwNzI2NDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mZmEzOWFmZGY1MDllMDQwZGJmMWIxM2RlZDg3NmNiZDk0ZTBmNzAwODYxY2QwZTdkNTU3MWFkNmVkMjYzY2FhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.svnq01DzFbgZLKtzqWN68yN8aHNiGFjYs-0eXhdRRwI)
> 
> Also you seem to have enough computer RAM but your GPU probably isn't powerful enough to run the Ollama model you've chosen on its own. To check, run the app and then in a different terminal put in the command
> 
> ```shell
> ollama ps
> ```
> 
> which will output how much of the model is running on your GPU vs CPU. Ideally you want 100% GPU but if some of it is on the CPU as well that means you're GPU can't run the model on its own.







How to run ollama models on gpu ? It's always running on my cpu
It's showing 100%for CPU

**Comment by leapingturtlefrog** [comment]
>I pressed q many times and q is not displayed

Try running the app, inputting a prompt, ex. `@Why is the sky blue?`, waiting a bit for it to scrape, and then pressing `q` only once, followed by `Ctrl+z`, even if the `q` does not show up. Then wait up to 2 minutes before pressing `q` and `Ctrl+z` again.

>How to run ollama models on gpu ? It's always running on my cpu
>It's showing 100%for CPU

What model are you using? Try a different model like `ollama run llama3.2` or `ollama run llama3.2:1b`. If that doesn't work, you can try this:
>Here's how to disable integrated graphics on Windows 11: Press Windows + X to open the "Power User Menu" Select Device Manager Double-click Display adapters to open the drop-down menu Right-click on the integrated graphics Select Disable device Click Yes to confirm

This has to do with Ollama, not this app.

**Comment by gokulcoder7** [comment]
For llama 3.1 8b , phi 3 models gpu is not at all used

**Comment by synth-mania** [comment]
@gokulcoder7 That is an Olama configuration issue. You might not have a compatible graphics card, I'm not sure.

**Comment by leapingturtlefrog** [comment]
>For llama 3.1 8b , phi 3 models gpu is not at all used

Those are still big models for your GPU, try
```cmd
ollama run ben1t0/tiny-llm
```
as a test case, and then check your GPU usage.

If that doesn't work, try uninstalling and reinstalling ollama. And it is also very important to make sure you have CUDA and Nvidia drivers installed, like this:
<img width="260" alt="image" src="https://github.com/user-attachments/assets/82fdd633-f615-4a9a-a562-3bf2831eb5d0">
[https://www.nvidia.com/en-us/drivers/details/232934/](https://www.nvidia.com/en-us/drivers/details/232934/)

**Comment by synth-mania** [comment]
Apologies @gokulcoder7 I just saw that you said you have a 3060. Even then, you should be able to run models like Phi3:3.8b, I haven't ever had to think about how much VRAM those little models use, but if it's too much, you could use a smaller quantization. As @leapingturtlefrog noted, there are also even smaller models out there to experiment with. Ollama should be an easy thing to get squared away, so let us know if you have any more issues.

**Comment by TheBlewish** [comment]
Hey look I made this program, I may or may not be very intoxicated on whiskey rn, but here is some chatGPT advice:

To help with GPU utilization in Ollama:

    Check your GPU Compatibility: Ensure your RTX 3060 is supported and that drivers are up-to-date.
    Install CUDA Toolkit: Install the appropriate CUDA version. You can get it from [NVIDIA's CUDA Toolkit page](https://developer.nvidia.com/cuda-toolkit).
    Verify Ollama's GPU Usage:
        Run ollama ps in a terminal while the model is running to check resource allocation (GPU vs CPU usage).
    Set GPU for Ollama:
        Explicitly set the environment variable for GPU use:

        export CUDA_VISIBLE_DEVICES=0

        Replace 0 with the correct GPU ID if you have multiple GPUs.
    Test Smaller Models: Larger models might not fit in the 3060's VRAM. Try a smaller model like ben1t0/tiny-llm to confirm GPU functionality.

If you need more help, you can also check Ollama's documentation or open a new issue with details about your configuration.



That was CHATGPT ADVICE for ya mate! hopefully it helps if now send a query back and I am sure these fine people such as SYNTH-MANIA (BTW MATE thanks so much for your activity in this project, you really have contributed a lot of time and effort genuinely good on ya! synth-mania! I struggle a lot with this workload and you genuinely do help out a lot with the issues sections and stuff so thanks mate!) will help ya out!

**Comment by TheBlewish** [comment]
sorry for my unproffessional response earlier, but based on the issue, and you saying you running the latest commit, I think your not using the windows branch

https://github.com/TheBlewish/Automated-AI-Web-Researcher-Ollama/tree/feature/windows-support

Are you using this? If not then try it, cause the main doesn't work on windows, if it's still an issue please reopen this otherwise i'll close it under this assumption, if your on the windows branch reopen and i'll do my best to help ya out!

**Comment by synth-mania** [comment]
> Hey look I made this program, I may or may not be very intoxicated on whiskey rn, but here is some chatGPT advice:
> 
> To help with GPU utilization in Ollama:
> 
>     Check your GPU Compatibility: Ensure your RTX 3060 is supported and that drivers are up-to-date.
>     Install CUDA Toolkit: Install the appropriate CUDA version. You can get it from [NVIDIA's CUDA Toolkit page](https://developer.nvidia.com/cuda-toolkit).
>     Verify Ollama's GPU Usage:
>         Run ollama ps in a terminal while the model is running to check resource allocation (GPU vs CPU usage).
>     Set GPU for Ollama:
>         Explicitly set the environment variable for GPU use:
> 
>         export CUDA_VISIBLE_DEVICES=0
> 
>         Replace 0 with the correct GPU ID if you have multiple GPUs.
>     Test Smaller Models: Larger models might not fit in the 3060's VRAM. Try a smaller model like ben1t0/tiny-llm to confirm GPU functionality.
> 
> If you need more help, you can also check Ollama's documentation or open a new issue with details about your configuration.
> 
> 
> 
> That was CHATGPT ADVICE for ya mate! hopefully it helps if now send a query back and I am sure these fine people such as SYNTH-MANIA (BTW MATE thanks so much for your activity in this project, you really have contributed a lot of time and effort genuinely good on ya! synth-mania! I struggle a lot with this workload and you genuinely do help out a lot with the issues sections and stuff so thanks mate!) will help ya out!

High praise lol, I appreciate it. Professionalism isn't a strict requirement in side projects and solo efforts anyways.

(I might be relaxing in the same manner here after finals week, depending on the outcome, after all)

Also in part due to a last minute scramble to get stuff done before the end of my semester, I have not been as active as I was last week, but I'm genuinely interested in this project and I'm definitely going to continue to help out.

;)

**Comment by blunweon** [comment]
For making use of GPU instead of only CPU, check out my issue #74. I managed to add some parameters to the python script and managed to get GPU offloading to work.
